src/components/Form/Form.tsx

```diff
@@ -55,7 +58,13 @@ const Form = <T extends FormDetails>({
       dispatch({ type: "set_form", form });
       setIsLoading(false);
     });
-  }, [asyncGetDefaults]);
+  }, [asyncGetDefaults, dispatch]);
+
+  useEffect(() => {
+    const hasChanges =
+      JSON.stringify(state.form) !== JSON.stringify(formVals.current);
```

Comment:
Let's not do this, comparing like this isn't always reliable.

Reply:
liek already said in previous review, use shadcn's form

https://ui.shadcn.com/docs/components/form